src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker_Binding.cs

```diff
@@ -0,0 +1,392 @@
+
+        private bool TrySimplifyBaseAccessExpression(MemberAccessExpressionSyntax node)
+        {
+            if (node.Expression.Kind() != SyntaxKind.BaseExpression)
+                return false;
+
+            // We have `base.SomeMember(...)`.  This is potentially simplifiable to `SomeMember`.
+            // However, this involves complex overload-instance-resolution logic.  So we fall-back
+            // to the full simplification analysis system.
+            return TrySimplify(node);
+        }
+
+        private bool TrySimplifyStaticMemberAccessInScope(MemberAccessExpressionSyntax node, string memberName)
+        {
+            // see if we can just access this member using it's name alone here.
+            if (!_staticNamesInScope.Contains(memberName))
+                return false;
+
+            return TrySimplify(node);
+        }
+
+        private bool TrySimplifyStaticMemberAccessThroughDerivedType(MemberAccessExpressionSyntax node)
+        {
+            // We have `Y.Z`.  It might be that `Z` is a static member actually declared through a
+            // base-type of `Y` called `X`.  This can be simplified to `X.Z`.
+            //
+            // To avoid having to call this on every `A.B` member access, we ensure that
+            // there's actually a type called `A` somewhere in our compilation. This helps
+            // avoid costly binding in many cases.
+
+            var exprName = node.Expression.GetRightmostName();
+            if (!_compilationTypeNames.Contains(exprName.Identifier.ValueText!))
+                return false;
+
+            // Member on the right of the dot needs to be a static member or another named type.
+            var nameSymbol = _semanticModel.GetSymbolInfo(node.Name).Symbol;
+            if (!IsNamedTypeOrStaticSymbol(nameSymbol))
+                return false;
+
+            if (nameSymbol.ContainingType == null)
+                return false;
+
+            // Don't simplify a container into a generic type.  Users do not consider it to actually
+            // be simpler to go from D.P to B<X>.P
+            if (nameSymbol.ContainingType.Arity > 0)
+                return false;
+
+            // If the user is already accessing the static through its containing type, there's
+            // nothing we need to simplify to.
+            var containerSymbol = _semanticModel.GetSymbolInfo(node.Expression).Symbol;
+            if (Equals(containerSymbol, nameSymbol.ContainingType))
+                return false;
+
+            return TrySimplify(node);
+        }
+
+        public override void VisitQualifiedCref(QualifiedCrefSyntax node)
+        {
+            // A qualified cref could be many different types of things.  It could be referencing a
+            // namespace, type or member (including static and instance members).  Because of this
+            // we basically have no avenues for bailing early and we have to try out all possible
+            // simplifications paths.
```

Comment:
note: we could bail out earlier if we store instance members in our indices as well as static members.